src/Dumplie/Application/Metadata/Schema/Field/AssociationField.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Metadata\Schema\Field;
+
+use Dumplie\Application\Exception\Metadata\InvalidArgumentException;
+use Dumplie\Application\Exception\Metadata\InvalidValueException;
+use Dumplie\Application\Metadata\Hydrator;
+use Dumplie\Application\Metadata\Metadata;
+use Dumplie\Application\Metadata\Schema\FieldDefinition;
+use Dumplie\Application\Metadata\HydrationData;
+use Dumplie\Application\Metadata\Schema\Type;
+use Dumplie\Application\Metadata\Schema\TypeSchema;
+
+final class AssociationField implements FieldDefinition
+{
+    const NAME = 'association';
+
+    /**
+     * @var TypeSchema
+     */
+    private $typeSchema;
+
+    /**
+     * @var bool
+     */
+    private $nullable;
+
+    /**
+     * @var array
+     */
+    private $options = [];
+
+    /**
+     * @param TypeSchema $typeSchema
+     * @param bool        $nullable
+     * @param array       $options
+     */
+    public function __construct(TypeSchema $typeSchema, bool $nullable = false, array $options = [])
+    {
+        $this->typeSchema = $typeSchema;
+        $this->nullable = $nullable;
+        $this->options = $options;
+    }
+
+    /**
+     * @return string
+     */
+    public function name() : string
+    {
+        return self::NAME;
+    }
+
+    /**
+     * @return Type
+     */
+    public function type() : Type
+    {
+        return Type::association();
+    }
+
+    /**
+     * @return array
+     */
+    public function options() : array
+    {
+        return $this->options;
+    }
+
+    /**
+     * @return null
+     */
+    public function defaultValue()
+    {
+        return;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isNullable() : bool
+    {
+        return $this->nullable;
+    }
+
+    /**
+     * @return TypeSchema
+     */
+    public function typeSchema()
+    {
+        return $this->typeSchema;
+    }
+
+    /**
+     * @param $value
+     * @return bool
+     */
+    public function isValid($value) : bool
+    {
+        if ($value === null && $this->nullable) {
+            return true;
+        }
+
+        return $value instanceof Metadata;
+    }
+
+    /**
+     * @param $value
+     * @return string
+     * @throws InvalidValueException
+     */
+    public function serialize($value) : string
+    {
+        if (!$value instanceof Metadata) {
+            throw InvalidValueException::valueDoesNotMatchType($this, $value);
+        }
+
+        return $value->id();
+    }
+
+    /**
+     * @param string $serializedValue
+     *
+     * @return null|HydrationData
+     * @throws InvalidArgumentException
+     */
+    public function deserialize($serializedValue)
+    {
+        if (null === $serializedValue && $this->isNullable()) {
+            return null;
+        }
+
+        if (!is_callable($serializedValue)) {
+            throw InvalidArgumentException::expected('callable', $serializedValue);
+        }
+
+        // TODO - make it lazy
```

Comment:
Its just a reminder. The plan is to change `HydratorData` so it accepts not only arrays and then `DefaultHydrator` becomes `LazyHydrator`.